packages/core/saved-objects/core-saved-objects-migration-server-internal/src/zdt/next.ts

```diff
@@ -56,3 +95,13 @@ export const next = (context: MigratorContext) => {
     }
   };
 };
+
+const createDelayFn =
+  (state: State) =>
```

Comment:
NIT we could update the function signature to simply accept a number.
Also, the same code block exists for the "v2 next.ts", so perhaps we can factorise this.

Reply:
Extracted the function in https://github.com/elastic/kibana/pull/152219/commits/6d0b3f0de4fa3d631a728d03ef6d29552bb0e590. Kept passing the full state as it felt more consistent, but we can revisit if we're planning on using the HOC for other things